### PR TITLE
DAOS-12974 test: nvme/pool_capacity.py - Use threads to create contai…

### PR DESCRIPTION
**Pull request by shimizukko**
…ners

The test creates 50 containers for each of the 10 pools 20 times (10 x 50 x 20). Creating many containers serially takes significant amount of time, so use threads to create the containers in parallel.

Also reduce the number of iterations from 20 to 2 and reduce the timeout.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_nvme_pool_capacity
Skip-func-hw-test-medium-md-on-ssd: false

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — DISMISSED** [review]
LGTM. For further improvement, have we considered whether we really need to run in a loop 20 times?
https://github.com/daos-stack/daos/blob/ff067d653b5e2f570b78e0f04ec93a5acf37ef5c/src/tests/ftest/nvme/pool_capacity.py#L229C14-L229C36

File: src/tests/ftest/nvme/pool_capacity.py [hunk 1]
```diff
@@ -94,16 +94,28 @@ def run_test_create_delete(self, num_pool=2, num_cont=5, total_count=100):
...
+                        properties="reclaim:disabled"))
 
                 display_string = "pool{} space at the Beginning".format(val)
                 self.pool[-1].display_pool_daos_space(display_string)
                 nvme_size_begin[val] = self.pool[-1].get_pool_free_space("NVME")
-                # To be fixed by DAOS-12974
+                # Before creating a container, we'll disconnect the pool. However, disconnecting the
+                # pool with threads causes the test to crash, so disconnect before creating any
+                # container. We connect during get_pool_free_space() above.
+                self.pool[-1].disconnect()
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/pool_capacity.py:108]: I think it's worth re-verifying this after #14153 lands.

Comment (shimizukko) [reply src/tests/ftest/nvme/pool_capacity.py]: I removed the disconnect and it works. Thanks.

**Comment by shimizukko** [comment]
> LGTM. For further improvement, have we considered whether we really need to run in a loop 20 times? https://github.com/daos-stack/daos/blob/ff067d653b5e2f570b78e0f04ec93a5acf37ef5c/src/tests/ftest/nvme/pool_capacity.py#L229C14-L229C36

I reduced the iteration to 2 and the test time was reduced from 60 min to 16 min.

**Review by daltonbohning — DISMISSED** [review]

**Comment by shimizukko** [comment]
> > > LGTM. For further improvement, have we considered whether we really need to run in a loop 20 times? https://github.com/daos-stack/daos/blob/ff067d653b5e2f570b78e0f04ec93a5acf37ef5c/src/tests/ftest/nvme/pool_capacity.py#L229C14-L229C36
> > 
> > 
> > I reduced the iteration to 2 and the test time was reduced from 60 min to 16 min.
> 
> Can we reduce the test timeout then?

Done

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Pull request merged**